### PR TITLE
[DT 1067] Added ud client as proxy server and remove default rpc url

### PR DESCRIPTION
**Pull request by tunguyennnnn**
## Background

This release offers a new interface for resolution library to call unstoppable domains proxy service for resolving onchain data.

## Changes

* Add new interface to configure interacting with proxy service
* Remove the use of default Infura
  * Migrate to use environment variables  for unit test
  * Remove configuration of default RPC provider

## Code Review

This Pull Request was thoroughly reviewed and meets all Code Review Standards pertaining to:

- [ ] **Implementation** - satisfied acceptance criteria
- [ ] **Communication** - notified engineers/stakeholders about impactful changes
- [ ] **Error handling** - handled, logged & alerted on errors
- [ ] **Documentation** - wrote readable code & commits, documented unintuitive code
- [ ] **Testing** - unit & E2E test coverage, tested in staging, DB migrations backwards compatible
- [ ] **Security** - sanitized inputs, vetted dependencies, validated/secured API requests, no committed secrets
- [ ] **Performance** - optimized expensive algorithms & database queries

Please select all **applied** standards relevant to this PR.
## Confirmation

- [ ] **Assignee Confirmation** - I (the author) have filled out the template above
- [ ] **Reviewer Confirmation** - I (a/the reviewer) confirm 1) the author has filled out the template and checked the box that says **Assignee Confirmation** 2) I reviewed the code and selected all applicable items in the code review section.

### CODE REVIEW

File: udclient/udclient.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type UdClient struct {
+	L1ContractBackend *ethclient.Client
+	L2ContractBackend *ethclient.Client
+}
+
+var libVersion = "resolution-go-v2.3.2"
+var proxyBaseUrl = "https://resolve.unstoppabledomains.com"
```
Comment (sammyluo) [inline udclient/udclient.go:19]: Shall we just use the new URL with API Gateway?

File: udclient/udclient.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+// Dial connects a client to the a proxy service with a authentication key
+func Dial(apiKey string) (*UdClient, error) {
+	l1ProxyUrl := proxyBaseUrl + l1RpcProxyPath
+	l2ProxyUrl := proxyBaseUrl + l2RpcProxyPath
+
+	tokenHeader := rpc.WithHeader("authorization", "Bearer "+apiKey)
+
+	libClientHeaderString := libVersion + "/" + runtime.Version()
+	agentHeader := rpc.WithHeader("x-lib-client", libClientHeaderString)
```
Comment (sammyluo) [inline udclient/udclient.go:31]: In the JS PR, you named this header as `X-Lib-Agent` instead of `x-lib-client`. Let's pick one and make them consistent.

**Review by sammyluo — CHANGES_REQUESTED** [review]

File: udclient/udclient.go (hunk 1 above)
Comment (tunguyennnnn) [reply udclient/udclient.go:19]: 👍 yea I was keeping this for testing. will update it!

File: udclient/udclient.go (hunk 2 above)
Comment (tunguyennnnn) [reply udclient/udclient.go:31]: yea will pick `X-Lib-Agent` 👍

File: udclient/udclient.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+// Dial connects a client to the a proxy service with a authentication key
+func Dial(apiKey string) (*UdClient, error) {
+	l1ProxyUrl := proxyBaseUrl + l1RpcProxyPath
+	l2ProxyUrl := proxyBaseUrl + l2RpcProxyPath
+
+	tokenHeader := rpc.WithHeader("Authorization", "Bearer "+apiKey)
+
+	libClientHeaderString := libVersion + "/" + runtime.Version()
+	agentHeader := rpc.WithHeader("X-Lib-Client", libClientHeaderString)
```
Comment (sammyluo) [inline udclient/udclient.go:31]:
```suggestion
	agentHeader := rpc.WithHeader("X-Lib-Agent", libClientHeaderString)
```

**Review by sammyluo — CHANGES_REQUESTED** [review]

**Review by sammyluo — APPROVED** [review]
Nice! 🚀 ⭐

**Pull request merged**